### ISSUE TITLE
STM32F303CCTx

### ISSUE THREAD

**Issue by Timmotools** [skuep/AIOC#80]
Hi,

Love your work here and I have a board coming soon. 

I have a question though. Would the firmware run on the STM32F303CCTx chip? 

While I am waiting, I was going through my junk boxes and found an old drone SPracing F3 flight controller. These have the STM32F303CCTx chip. 

I have mapped out all of the pins needed for the mappings and all the pins I need are going to solder pads. So little hardware modding and adding a few components. It should work. It's also running 8mhz clock already.

Thanks.
![20240721_081020](https://github.com/user-attachments/assets/f03343d7-fdfb-49bd-af43-a4f901071646)
![20240721_081027](https://github.com/user-attachments/assets/aa553ee2-76d9-47f5-b29f-2e7221b976db)

**Comment by Timmotools** [comment]
Just an update on this. I have flashed it with the latest firmware with stmcube via the built in usb to UART connection usually used for flashing or configuring cleanflight or betaflight. Then wired up the usb interface with a scrap usb lead. I can see in dev manager the sound and serial interfaces. 

This looks promising. 

It's just after midnight here in Australia and I have to work in the morning. Tomorrow I'll have a play around with the rest of the extra components and see if it works. 

Cheers
VK3FTP.

**Comment by skuep** [comment]
Nice! I think as long as the chip has an ADC and DAC, you should be golden. Remember to select the correct part for the compilation process and you might need to swap the linker script for your MCU.

But if you're already seeing the USB devices, you should be almost there 😁 you will need some circuitry for the analog stuff though (level shifting, ac coupling and signal attenuation). Also for PTT it's best of you also use some simple bipolar NPN transistors.

Just a heads up, in the past I had a noisy signal at 1khz due to the USB interface on the MCU being on the same power supply, which is why I chose a dual regulator with one output for analog and one for digital power.

**Comment by Timmotools** [comment]
Just an update on this. I have built all of the level converters and ptt control lines (single ptt) as per your schematic. The audio in/out and ptt are working fine. However the serial lines are not.

I did have to remove a 3.3v pull-up and short a series resistor on the AF-OUT (vbat) line.

Here is a photo of the board. Definitely not pretty but it's not a permanent solution. More a proof of concept.

![20240724_234701](https://github.com/user-attachments/assets/70d2ee29-ab31-41dd-9acd-3fe7eea7f281)

I tried compiling the firmware with my chip selected but same issue. Chirp says short header.
On some more research on the board I found that the USART1 line is also the same usart line used for the built on usb to UART chip. So this is probably causing issues. 

I'll change the code and try later.
